cmd/kubeadm/app/preflight/checks_test.go

```diff
@@ -703,3 +706,95 @@ func TestSetHasItemOrAll(t *testing.T) {
 		}
 	}
 }
+
+func TestExternalEtcdVersionCheck(t *testing.T) {
+	tests := []struct {
+		name           string
+		response       string
+		serverStatus   int
+		expectErr      bool
+		expectedErrMsg string
+	}{
+		{
+			name:     "supported etcd version",
+			response: fmt.Sprintf(`{"etcdserver":"%s","etcdcluster":"%s"}`, kubeadmconstants.MinExternalEtcdVersion, kubeadmconstants.MinExternalEtcdVersion),
+		},
+		{
+			name:           "unsupported etcd version",
+			response:       `{"etcdserver":"3.0.0","etcdcluster":"3.0.0"}`,
+			expectErr:      true,
+			expectedErrMsg: fmt.Sprintf("this version of kubeadm only supports external etcd version >= %s. Current version: 3.0.0", kubeadmconstants.MinExternalEtcdVersion),
+		},
+		{
+			name:           "404 response from server",
+			serverStatus:   http.StatusNotFound,
+			expectErr:      true,
+			expectedErrMsg: "server responded with non-successful status: 404 Not Found",
+		},
+		{
+			name:           "500 response from server",
+			serverStatus:   http.StatusInternalServerError,
+			expectErr:      true,
+			expectedErrMsg: "server responded with non-successful status: 500 Internal Server Error",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Given that you are just surfacing simple errors, and we generally discourage stub'd test servers vs. real in memory servers in the code, you can trim this down to just the changes that at @dixudx had requested.